pkg/forwarder/sync_forwarder.go

```diff
@@ -25,9 +25,11 @@ type SyncForwarder struct {
 }
 
 // NewSyncForwarder returns a new synchronous forwarder.
-func NewSyncForwarder(domainResolvers map[string]resolver.DomainResolver, timeout time.Duration) *SyncForwarder {
+func NewSyncForwarder(domainResolvers map[string]resolver.DomainResolver, timeout time.Duration, isServerlessMode bool) *SyncForwarder {
```

Comment:
Right now I think the SyncForwarder is **only** used for the Lambda Extension. So, maybe we don't need the `isServerlessMode` argument here yet.

Reply:
Yes this is currently the case, I just wanted to be explicit here as I feel like if another non serverless-related use case would need a SyncForwarder, this do-not-reuse-tcp connexion will be a severe, most likely unwanted and hidden side effect?

Reply:
Perhaps calling it "closeConnections" argument would make more sense?  The serverless agent has a lot of "if serverless" spread throughout the codebase right now, and that makes understanding the code pretty difficult.  I think it would be easier to understand the behavior specific to the component in question (in this case, closing connections immediately) and then apply that behavior at the edge.

In a parallel case, there is some serverless-specific stuff in `pkg/logs/{logs,agent}.go` which I am working on slowly factoring out so that both cmd/agent and cmd/serverless can call `logs.New()` to create an agent and then `AddScheduler`, `AddLauncher`, etc. as appropriate for the agent flavor.  Then there would be no mention of "serverless" in the logs agent.